src/material/core/typography/_all-typography.scss

```diff
@@ -43,6 +43,13 @@
     $config: mat-typography-config();
   }
 
+  // TODO: Do not use individual mixins. Instead, use the all-theme mixin and only
+  // specify a `typography` config while setting `color` and `density` to `null`. This needs
+  // discussion as it would introduce a circular dependency for typography because:
+  //   -> `all-theme` -> `mat-core` -> `all-typography` -> `all-theme`.
+  // This happens because developers rely on the `mat-core` to be available when
+  // the `all-theme` file is imported.
```

Comment:
A possible solution would be that developers need to explicitly import the file that exposes the `mat-core` mixin. We can also keep it as is for now and import the individual typography mixins.